calico_kubernetes/calico_kubernetes.py

```diff
@@ -181,32 +227,43 @@ def _container_add(self, pid, ip, interface, hostname, orchestrator_id):
         # Obtain information from Docker Client and validate container state
         self._validate_container_state(self.docker_id)
 
-        # Assign ip address through IPAM Client
-        try:
-            ip_assigned = self._datastore_client.assign_address(None, ip)
-        except PoolNotFound:
-            logger.error("IP address %s does not belong to any configured "
-                         "pools. Exiting.", ip)
-            sys.exit(1)
+        if CALICO_IPAM == 'calico':
+            logger.info("Using Calico IPAM")
+            # Assign ip address through IPAM Client
+            try:
+                ip_lists = self._datastore_client.auto_assign_ips(
+                1, 0, str(self.docker_id), None)
+            except RuntimeError as err:
+                logger.error("Cannot auto assign IPAddress: %s", err.message)
+
+            # Concatenate lists into one single list
+            ip_list = ip_lists[0]+ip_lists[1]
```

Comment:
I don't think we need to concat these - we don't care about the IPv6 addresses for now, so lets just throw them away.